fiftyone/core/utils.py

```diff
@@ -1539,7 +1569,140 @@ def _compute_batch_size(self):
+            callback function to invoke instead
+        total (None): the length of ``iterable``. Only applicable when
+            ``progress=True``. If not provided, it is computed via
+            ``len(iterable)``, if possible
+        transform_fn (None): a transform function to apply to each item of the batch
+    """
+
+    def __init__(
+        self,
+        iterable,
+        size_calculation_fn=None,
+        target_size=2**20,
+        max_batch_size=None,
+        progress=False,
+        total=None,
+        transform_fn=None,
+    ):
+        super().__init__(
+            iterable,
+            return_views=False,
+            progress=progress,
+            total=total,
+            transform_fn=transform_fn,
+        )
+        self.max_batch_size = max_batch_size
+        self.target_size = target_size
+        self._prev_element = None
+        self._size_calculation_fn = (
+            size_calculation_fn
+            if size_calculation_fn
+            else default_calculate_size
+        )
+
+    def __iter__(self):
+        super().__iter__()
+        try:
+            self._prev_element = next(self._iter)
+        except StopIteration:
+            # If iterable is empty, we want to throw StopIteration at the first
+            #   call to next(), not here.
+            self._prev_element = None
+        return self
+
+    def __next__(self):
+        if self._render_progress and self._last_batch_size:
+            self._pb.update(count=self._last_batch_size)
+
+        if self._prev_element is None:
+            raise StopIteration()
+
+        batch_content_size = self._size_calculation_fn(self._prev_element)
+        curr_batch_size = 1
+        curr_batch = [self._prev_element]
+
+        while True:
+            try:
+                self._prev_element = next(self._iter)
+                next_element_size = self._size_calculation_fn(
+                    self._prev_element
+                )
+                if (
+                    self.max_batch_size
+                    and curr_batch_size >= self.max_batch_size
+                ) or (
+                    batch_content_size + next_element_size > self.target_size
+                ):
+                    break
+                curr_batch_size += 1
+                batch_content_size += next_element_size
+                curr_batch.append(self._prev_element)
+            except StopIteration:
+                self._prev_element = None
+                break
+        self._last_batch_size = curr_batch_size
+        return curr_batch
+
+
+def default_calculate_size(obj):
+    try:
+        obj = (
+            obj.to_mongo_dict(include_id=True)
+            if hasattr(obj, "to_mongo_dict")
+            else obj
+        )
+        return len(json_util.dumps(obj))
+    except Exception:
+        return len(str(obj))
+
+
+def get_default_batcher(
+    iterable,
+    progress=False,
+    total=None,
+    size_calc_fn=default_calculate_size,
```

Comment:
I think this can default None, since the default is handled in ContentSizeBatcher init, yes?

Also, its missing from docstring